onnxruntime/test/autoep/library/example_plugin_ep_kernel_registry/ep.cc

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "ep.h"
+
+#include <gsl/span>
+#include <array>
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "ep_factory.h"
+#include "../plugin_ep_utils.h"
+
+ExampleKernelEp::ExampleKernelEp(ExampleKernelEpFactory& factory, const OrtLogger& logger)
+    : OrtEp{},  // explicitly call the struct ctor to ensure all optional values are default initialized
+      factory_{factory},
+      ort_api_{factory.GetOrtApi()},
+      ep_api_{factory.GetEpApi()},
+      name_{factory.GetEpName()},
+      logger_{logger} {
+  ort_version_supported = ORT_API_VERSION;  // set to the ORT version we were compiled with.
+
+  // Initialize the execution provider's function table
+  GetName = GetNameImpl;
+  GetCapability = GetCapabilityImpl;
+  GetKernelRegistry = GetKernelRegistryImpl;
+
+  // This is not a compiling EP, so don't need the following
+  Compile = nullptr;
+  ReleaseNodeComputeInfos = nullptr;
+
+  IGNORE_ORTSTATUS(ort_api_.Logger_LogMessage(&logger_,
+                                              OrtLoggingLevel::ORT_LOGGING_LEVEL_INFO,
+                                              ("ExampleKernelEp has been created with name " + name_).c_str(),
+                                              ORT_FILE, __LINE__, __FUNCTION__));
+}
+
+ExampleKernelEp::~ExampleKernelEp() = default;
+
+/*static*/
+const char* ORT_API_CALL ExampleKernelEp::GetNameImpl(const OrtEp* this_ptr) noexcept {
+  const auto* ep = static_cast<const ExampleKernelEp*>(this_ptr);
+  return ep->name_.c_str();
+}
+
+/*static*/
+OrtStatus* ORT_API_CALL ExampleKernelEp::GetCapabilityImpl(OrtEp* this_ptr, const OrtGraph* ort_graph,
+                                                           OrtEpGraphSupportInfo* graph_support_info) noexcept {
+  try {
+    ExampleKernelEp* ep = static_cast<ExampleKernelEp*>(this_ptr);
+
+    Ort::ConstGraph graph{ort_graph};
+    std::vector<Ort::ConstNode> all_nodes = graph.GetNodes();
+
+    if (all_nodes.empty()) {
+      return nullptr;  // No nodes to process
+    }
+
+    // Collect candidate nodes that this EP may support.
+    std::vector<Ort::ConstNode> candidate_nodes;
+
+    for (const auto& node : all_nodes) {
+      std::string op_type = node.GetOperatorType();
+
+      if (op_type == "Relu" || op_type == "Squeeze") {
+        candidate_nodes.push_back(node);
+      } else if (op_type == "Mul") {
+        std::vector<Ort::ConstValueInfo> inputs = node.GetInputs();
+        assert(inputs.size() == 2);
```

Comment:
nit: we run ONNX type and shape inferencing before the EP sees the model, so this sort of assert is unnecessary. as it's example code that could be misleading as it might make an EP author think they need to do all those validity checks. 

could alternatively add a comment calling out that the ONNX type/shape inferencing validations have been applied previously.

Reply:
Removed